### PR TITLE
Remove 2.6.32 and 2.6.38 branches

### PR DESCRIPTION
**Pull request by Pinglinux**
RHEL 6 was EOL last year, which was based on 2.6.32. Plus most Linux
distributions do not support 2.6.38 any more.

Signed-off-by: Ping Cheng <ping.cheng@wacom.com>

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1,6 +1,6 @@
 AC_PREREQ(2.60)
 AC_INIT([input-wacom],
-	[0.47.0])
+	[0.48.0])
```
Comment (jigpu) [inline configure.ac]: The version number should only be updated when we make a release.

File: configure.ac [hunk 2]
```diff
@@ -217,31 +205,15 @@ struct power_supply_desc test;
...
-])
-
 dnl Check which version of the driver we should compile
 AC_DEFUN([WCM_EXPLODE], [$(echo "$1" | awk '{split($[0],x,"[[^0-9]]"); printf("%03d%03d%03d\n",x[[1]],x[[2]],x[[3]]);}')])
 EXPLODED_VER="WCM_EXPLODE($MODUTS)"
 if test "$EXPLODED_VER" -lt "WCM_EXPLODE(2.6.32)"; then
-	AC_MSG_ERROR([use the kernel driver included in the 'linuxwacom' package for kernels older than 2.6.32])
-elif test "$EXPLODED_VER" -lt "WCM_EXPLODE(2.6.38)"; then
-	WCM_KERNEL_VER="2.6.32"
+	AC_MSG_ERROR([Please use the kernel driver included in the 'linuxwacom' package for kernels older than 2.6.32])
```
Comment (jigpu) [inline configure.ac]: We should remove this check and error message. The error message below for kernels older than 3.7 is sufficient.

File: configure.ac [hunk 3]
```diff
@@ -217,31 +205,15 @@ struct power_supply_desc test;
...
 elif test "$EXPLODED_VER" -lt "WCM_EXPLODE(3.7)"; then
-	WCM_KERNEL_VER="2.6.38"
+	AC_MSG_ERROR(m4_normalize([Kernels older than 3.7 is no longer supported by input-wacom.
```
Comment (jigpu) [inline configure.ac]: "is" -> "are"

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -9,18 +9,8 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        kernel: ["4.6", "4.5", "4.1", "3.19", "3.18", "3.17", "3.7", "3.3", "2.6.38", "2.6.36", "2.6.35", "2.6.32"]
+        kernel: ["4.6", "4.5", "4.1", "3.19", "3.18", "3.17", "3.7"]
```
Comment (jigpu) [inline .github/workflows/main.yml:12]: Please merge this entire commit with the previous one.

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -45,9 +35,9 @@ jobs:
           KERNELDIR=${HOME}/kernel
           KERNELROOT=${KERNELDIR}/linux-${KERNELVER}
 
-          # Generate the "v2.6" or "v{major}.x" URL path component from the kernel
+          # Generate the "v{major}.x" URL path component from the kernel
           # version and then download and unpack the sources
-          SERIES=v$(awk -vFS=. -vOFS=. '{if ($1!=2) {$2="x"} NF=2; print}' <<< ${KERNELVER})
+          SERIES=v$(awk -vFS=. -vOFS=. '{$2="x"; print}' <<< ${KERNELVER})
```
Comment (jigpu) [inline .github/workflows/main.yml:40]: We could leave `NF=2` here just to be safe, but I'll leave that decision up to you.

**Review by jigpu — CHANGES_REQUESTED** [review]
A couple minor nitpicks.

Also, just for future reference, here are a few of the other impacted distros (all of which are EOL):

| Distro | Kernel | EOL |
|-|-|-|
| Debian 7 | 3.2 | 2018-06-01 |
| RHEL 6 | 2.6.32 | 2020-11-30 |
| SuSE Linux Enterprise 11.3 | 3.0 | 2016-01-31 |
| SuSE Linux Enterprise 11.4 | 3.0 | 2019-05-31 |
| Ubuntu 12.04 ("Precise Pangolin") | 3.2 | 2017-04-28 |

**Comment by Pinglinux** [comment]
> A couple minor nitpicks.
> 
> Also, just for future reference, here are a few of the other impacted distros (all of which are EOL):
> 
> Distro	Kernel	EOL
> Debian 7	3.2	2018-06-01
> RHEL 6	2.6.32	2020-11-30
> SuSE Linux Enterprise 11.3	3.0	2016-01-31
> SuSE Linux Enterprise 11.4	3.0	2019-05-31
> Ubuntu 12.04 ("Precise Pangolin")	3.2	2017-04-28

Thank you. I'll add this info to the commit comments.

File: configure.ac (hunk 2 above)
Comment (Pinglinux) [reply configure.ac]: We can't remove the last line. It is still true that kernels older than 2.6.32 are supported by the old all-in-one linuxwacom package

Comment (Pinglinux) [reply configure.ac]: Well, probably no one cares about those old kernels anyway. I'll remove the message

**Comment by jigpu** [comment]
Thanks Ping; the changes look good :+1: Feel free to merge the changes now.

**Pull request merged**